### PR TITLE
Improve ArC integration

### PR DESCRIPTION
**Pull request by mkouba**
- merge FullArchiveIndexBuildItem into BeanArchiveBuildItem
- fix the default annotation transformer - use special stereotype
instead of Dependent

This a follow-up of https://github.com/quarkusio/quarkus/pull/1383

### CODE REVIEW

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanArchives.java [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    private static IndexView buildBuiltinIndex() {
+        Indexer indexer = new Indexer();
+        index(indexer, ActivateRequestContextInterceptor.class.getName());
+        return indexer.complete();
+    }
+
+    /**
+     * This wrapper is used to index JDK classes on demand.
+     */
+    static class IndexWrapper implements IndexView {
```
Comment (gsmet) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanArchives.java]: This is new, isn't it?

Is it exposed to the extensions (i.e. is this wrapper part of the `IndexView` we get in the extensions)?

Comment (mkouba) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanArchives.java]: It's not new. Previously it was a part of the `BeanDeployment` class. And the original purpose was to make JDK classes part of the index whenever ArC asks for a `ClassInfo` of a `DotName` that is not present in the provided index (for example a producer method returns a JDK `List` which is obviously not part of the app index but we need to analyze the corresponding `ClassInfo`). Of course, it works for non-JDK classes too.

Also it's package-private so it's not exposed. Anyway, extension can access this wrapper through the `IndexView` interface (after injecting `BeanArchiveIndexBuildItem`).

**Comment by mkouba** [comment]
`Connect to repository.jboss.org:80 timed out` again :-(. I'll try to re-run once again.

**Comment by mkouba** [comment]
```
[ERROR] Failed to execute goal on project quarkus-jdbc-mssql-runtime: Could not resolve dependencies for project io.quarkus:quarkus-jdbc-mssql-runtime:jar:999-SNAPSHOT: Failed to collect dependencies at com.microsoft.sqlserver:mssql-jdbc:jar:7.2.1.jre8: Failed to read artifact descriptor for com.microsoft.sqlserver:mssql-jdbc:jar:7.2.1.jre8: Could not transfer artifact com.microsoft.sqlserver:mssql-jdbc:pom:7.2.1.jre8 from/to jboss-public-repository-group (http://repository.jboss.org/nexus/content/groups/developer/): Connect to repository.jboss.org:80 timed out -> [Help 1]
```
Hm, timeout again. Let's try again for the last time. @gsmet @stuartwdouglas is this a known issue?

**Comment by gsmet** [comment]
@mkouba yes :/. I created a topic on Zulip so that we can discuss this. We need to find a solution as it's really annoying. I will open an internal issue to see if it's an issue with our Nexus or a connection problem.

**Review by gsmet — APPROVED** [review]
Looks good to me.

I launched a new CI run.

**Review by stuartwdouglas — APPROVED** [review]

**Comment by mkouba** [comment]
Yet another timeout. I'm going to run the full build locally and if it passes I'll merge this.

**Pull request merged**